lib/src/signed_video_h26x_auth.c

```diff
@@ -691,6 +691,39 @@ compute_gop_hash(signed_video_t *self, h26x_nalu_list_item_t *sei)
   return status;
 }
 
+/**
+ * Decodes the SEI message, retrieves necessary parameters for authentication, and computes the hash
+ * for authenticity.
+ */
+static svrc_t
+prepare_golden_sei(signed_video_t *self, h26x_nalu_list_item_t *sei)
+{
+  assert(self);
+  svrc_t status = SV_UNKNOWN_FAILURE;
+  sign_or_verify_data_t *verify_data = self->verify_data;
+  const size_t hash_size = verify_data->hash_size;
+  self->gop_state.has_sei = true;
+  SV_TRY()
+    // If an SEI item is found and it hasn't been decoded yet, decode it.
+    if (sei && !sei->has_been_decoded) {
```

Comment:
This should always be true if the SEI is a Golden SEI, so there is no need for an if-statement.